packages/aws-cdk-lib/aws-ecs/lib/base/service-managed-volume.ts

```diff
@@ -0,0 +1,310 @@
+import { Construct } from 'constructs';
+import * as ec2 from '../../../aws-ec2';
+import * as iam from '../../../aws-iam';
+import * as kms from '../../../aws-kms';
+import { Size, Token } from '../../../core';
+import { BaseMountPoint, ContainerDefinition } from '../container-definition';
+
+/**
+* Represents the Volume configuration for an ECS service.
+*/
+export interface ServiceManagedVolumeProps {
+  /**
+  * The name of the volume. This corresponds to the name provided in the ECS TaskDefinition.
+  */
+  readonly name: string;
+
+  /**
+  * Configuration for an Amazon Elastic Block Store (EBS) volume managed by ECS.
+  *
+  * @default - undefined
+  */
+  readonly managedEBSVolume?: ServiceManagedEBSVolumeConfiguration
+}
+
+/**
+* Represents the configuration for an ECS Service managed EBS volume.
+*/
+export interface ServiceManagedEBSVolumeConfiguration {
+  /**
+  * An IAM role that allows ECS to make calls to EBS APIs on your behalf.
+  * This role is required to create and manage the Amazon EBS volume.
+  *
+  * @default - automatically generated role.
+  */
+  readonly role?: iam.IRole;
+
+  /**
+   * Indicates whether the volume should be encrypted.
+   *
+   * @default - Default Amazon EBS encryption.
+   */
+  readonly encrypted?: boolean;
+
+  /**
+   * AWS Key Management Service key to use for Amazon EBS encryption.
+   *
+   * @default - When `encryption` is turned on and no `kmsKey` is specified,
+   * the default AWS managed key for Amazon EBS volumes is used.
+   */
+  readonly kmsKeyId?: kms.IKey;
+
+  /**
+   * The volume type.
+   *
+   * @default - ec2.EbsDeviceVolumeType.GP2
+   */
+  readonly volumeType?: ec2.EbsDeviceVolumeType;
+
+  /**
+   * The size of the volume in GiB.
+   *
+   * You must specify either `sizeInGiB` or `snapshotId`.
+   * You can optionally specify a volume size greater than or equal to the snapshot size.
+   *
+   * The following are the supported volume size values for each volume type.
+   *   - gp2 and gp3: 1-16,384
+   *   - io1 and io2: 4-16,384
+   *   - st1 and sc1: 125-16,384
+   *   - standard: 1-1,024
+   *
+   * @default - The snapshot size is used for the volume size if you specify `snapshotId`,
+   * otherwise this parameter is required.
+   */
+  readonly sizeInGiB?: Size;
```

Comment:
Size type allows specification other than GiB, so

```suggestion
  readonly size?: Size;
```